python-package/lightgbm/basic.py

```diff
@@ -1192,15 +1192,21 @@ def predict(
                 predict_type=predict_type,
             )
         elif isinstance(data, list):
-            try:
-                data = np.array(data)
-            except BaseException as err:
-                raise ValueError("Cannot convert data list to numpy array.") from err
+            if isinstance(data[0], Sequence):
+                try:
+                    data = np.concatenate([i[:] for i in data])
+                except BaseException as err:
+                    raise ValueError('Cannot convert Sequence list to numpy array.') from err
+            else:
+                try:
+                    data = np.array(data)
+                except BaseException as err:
+                    raise ValueError('Cannot convert data list to numpy array.') from err
             preds, nrow = self.__pred_for_np2d(
                 mat=data,
                 start_iteration=start_iteration,
                 num_iteration=num_iteration,
-                predict_type=predict_type,
+                predict_type=predict_type
```

Comment:
```suggestion
                predict_type=predict_type,
```

This change looks unrelated. Can you please revert it, to limit the size of the diff and noise in the git blame?